### PR TITLE
Graph ql review

### PR DESCRIPTION
**Pull request by jeremychoi**

### CODE REVIEW

File: document/4-Web_Application_Security_Testing/12-API_Testing/01-Testing_GraphQL.md [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+Assess GraphQL's security configuration, as well as other common vulnerabilities which might affect GraphQL deployments.
+
+## How to Test
+
+Testing GraphQL nodes is not very different than testing other API technologies. This Guide suggests following these steps:
+
+### Introspection Queries
+
+A little bit of background:  
+Introspection queries are the way GraphQL lets you ask it what queries it supports. This is a good starting point on the information gathering phase.
```
Comment (jeremychoi) [inline document/4-Web_Application_Security_Testing/12-API_Testing/01-Testing_GraphQL.md:29]: It'd be nice to briefly add what are the Introspection queries.

File: document/4-Web_Application_Security_Testing/12-API_Testing/01-Testing_GraphQL.md [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+              "isDeprecated": false,
+              "deprecationReason": null
+            }
+          ],
+          "possibleTypes": null
+                ...
+        {
+~~~
+
+Now use GraphQL Voyager to get a better look on the output:
```
Comment (jeremychoi) [inline document/4-Web_Application_Security_Testing/12-API_Testing/01-Testing_GraphQL.md:216]: need to add about what the GraphQL Voyager is?

File: document/4-Web_Application_Security_Testing/12-API_Testing/01-Testing_GraphQL.md [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+<img src=images/GraphiQL4.png width="300" />  
+
+The documentation contains the data types, mutations, and basically every piece of information you can extract using Introspection.
+
+#### Using GraphQL Playgrounds
+
+GraphQL Playgrounds is a GraphQL client, which can be used to test different queries, as well as dividing GraphQL IDEs into different playgrounds, grouped by theme or by assigning a name to them.  
+Much like GraphiQL Playgrounds can create the documentation for you, without the need to manually sending introspection and processing the response but with one great advantage, it doesn't need GraphiQL interface to be available.  
+Another upside for this tool, is that it works just by directing the tool to the GraphQL node via a URL (there is also the option of using it locally with the data file) and then the magic happens without any user interaction.  
+Another advantage of the tool is that it can be used to test for vulnerabilities directly, you don't need to use a proxy to send HTTP requests (Burp for example), and so you can use this tool in order to manually plat with GraphQL and use Burp for other, more advanced payloads.
```
Comment (jeremychoi) [inline document/4-Web_Application_Security_Testing/12-API_Testing/01-Testing_GraphQL.md:248]: manually plat -> manually play

**Review by jeremychoi — COMMENTED** [review]
A quick review is done content-wise.

**Review by OrAsaf — APPROVED** [review]

**Comment by OrAsaf** [comment]
Thank you for the comments, I've changed the file accordingly.

**Pull request closed without merging**